third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -607,22 +609,23 @@ impl Interpreter {
         addr: AccountAddress,
         ty: &Type,
     ) -> PartialVMResult<()> {
-        let resource =
-            match Self::load_resource(loader, data_store, gas_meter, addr, ty)?.move_from() {
-                Ok(resource) => {
-                    gas_meter.charge_move_from(
-                        is_generic,
-                        TypeWithLoader { ty, loader },
-                        Some(&resource),
-                    )?;
-                    resource
-                },
-                Err(err) => {
-                    let val: Option<&Value> = None;
-                    gas_meter.charge_move_from(is_generic, TypeWithLoader { ty, loader }, val)?;
-                    return Err(err);
-                },
-            };
+        let resource = match Self::load_resource(loader, data_store, gas_meter, addr, ty)?
+            .move_from()
+        {
+            Ok(resource) => {
+                gas_meter.charge_move_from(
+                    is_generic,
+                    TypeWithLoader { ty, loader },
+                    Some(&resource),
+                )?;
+                resource
+            },
+            Err(err) => {
+                let val: Option<&Value> = None;
+                gas_meter.charge_move_from(is_generic, TypeWithLoader { ty, loader }, val)?;
+                return Err(err.with_message(format!("Failed to move resource from {:?}", addr)));
```

Comment:
The resource name is actually hard because the Type doesn't have struct information right now. We need an extra call into the resolver to get this info. However if we land the loader PR we should be able to get this much easier.

Reply:
Isn’t there ty there? I’m not looking at this in my phone so cant click through to check the typing

Reply:
@movekevin, enum Type just has an index into a table for structs, so it's not terribly useful to print.  Assuming the bytecode offset is known, though, the user can look at a listing and see the type.

But, @runtian-zhou:  we could add a method to loader to convert a type to a string for this debugging.  You have loader here.